src/module.c

```diff
@@ -97,7 +97,25 @@ static void RG_AfterForkChild() {
 	process_is_child = true;
 }
 
-static void RegisterForkHooks() {
+static int _RenameGraphContext(RedisModuleCtx *ctx, int type, const char *event,
+							   RedisModuleString *key_name) {
+	if(type != REDISMODULE_NOTIFY_ALL) return REDISMODULE_OK;
+	if(strcasecmp(event, "RENAME") == 0) {
+		RedisModuleKey *key = RedisModule_OpenKey(ctx, key_name, REDISMODULE_WRITE);
+		if(RedisModule_ModuleTypeGetType(key) == GraphContextRedisModuleType) {
+			GraphContext *gc = RedisModule_ModuleTypeGetValue(key);
+			size_t len;
+			const char *new_name = RedisModule_StringPtrLen(key_name, &len);
+			GraphContext_SetName(gc, new_name);
+		}
+		RedisModule_CloseKey(key);
+	}
+	return REDISMODULE_OK;
+}
+
+static void RegisterForkHooks(RedisModuleCtx *ctx) {
+	RedisModule_SubscribeToKeyspaceEvents(ctx, REDISMODULE_NOTIFY_ALL, _RenameGraphContext);
```

Comment:
`REDISMODULENOTIFYALL: All events`
Gives us much more than we really need, please use:
`REDISMODULENOTIFYGENERIC: Generic commands like DEL, EXPIRE, RENAME`
As we're not interested in all events.

Reply:
done